src/Jupyter/Magic/AbstractMagic.cs

```diff
@@ -97,46 +80,63 @@ public static (string, Dictionary<string, string>) ParseInput(string input)
         {
             Dictionary<string, string> inputParameters = new Dictionary<string, string>();
 
-            var args = input.Split(null as char[], StringSplitOptions.RemoveEmptyEntries);
+            // This regex looks for four types of matches:
+            // 1. (\{.*\})
```

Comment:
Thanks for the detailed explanation of the RE in the comments.

Reply:
That was as much for myself as for anyone else! Otherwise there's no way that I would be able to reverse-engineer this regex a month from now. 😄